src/EnergyPlus/Furnaces.cc

```diff
@@ -12506,7 +12506,8 @@ namespace Furnaces {
             }
         }
 
-        if (GetCurrentScheduleValue(state, state.dataFurnaces->Furnace(FurnaceNum).SchedPtr) == 0.0) {
+        if ((GetCurrentScheduleValue(state, state.dataFurnaces->Furnace(FurnaceNum).SchedPtr) == 0.0) || state.dataHVACGlobal->TurnFansOff ||
+            (GetCurrentScheduleValue(state, state.dataFurnaces->Furnace(FurnaceNum).FanAvailSchedPtr) == 0.0 && !state.dataHVACGlobal->TurnFansOn)) {
```

Comment:
Wouldn't the TurnFansOff flag turn off constant fan mode? I'm pretty sure if the fan avail schedule value is > 0, these flags don't apply. I haven't thought about these flags in some time and could easily be wrong but I think you want?

    if ((GetCurrentScheduleValue(state, state.dataFurnaces->Furnace(FurnaceNum).SchedPtr) == 0.0) || 
        (GetCurrentScheduleValue(state, state.dataFurnaces->Furnace(FurnaceNum).FanAvailSchedPtr) == 0.0 &&
            state.dataHVACGlobal->TurnFansOff && !state.dataHVACGlobal->TurnFansOn)) {

Reply:
My understanding is if `TurnFansOff` is true, then the fan should be off, no matter what the schedule says. Likewise, if `TurnFansOn` is true, then the fan should be on, no matter what the schedule says. That's how I read the code in Fans.cc [here](https://github.com/NREL/EnergyPlus/blob/d642102ef1e1a084b175b4ddb261cf8bd1afafa4/src/EnergyPlus/Fans.cc#L183-L187) and [here](https://github.com/NREL/EnergyPlus/blob/d642102ef1e1a084b175b4ddb261cf8bd1afafa4/src/EnergyPlus/Fans.cc#L1708-L1709) for simple fan, and similar code for variable flow and onoff.

Reply:
That certainly looks like the logic used. I still question whether a constant fan can be turned off by an avail manager. the outdoor air unit looks like an outlier.

// turn on logic

    if ((GetCurrentScheduleValue(state, Fan(FanNum).AvailSchedPtrNum) > 0.0 || state.dataFans->LocalTurnFansOn) &&
        !state.dataFans->LocalTurnFansOff && MassFlow > 0.0) {

    if (((GetCurrentScheduleValue(state, FanCoil(FanCoilNum).SchedPtr) > 0.0 && GetCurrentScheduleValue(state, FanCoil(FanCoilNum).fanAvailSchIndex) > 0.0) ||
         state.dataHVACGlobal->ZoneCompTurnFansOn) && !state.dataHVACGlobal->ZoneCompTurnFansOff) {

    if (OpMode == ContFanCycCoil && GetCurrentScheduleValue(state, state.dataFurnaces->Furnace(FurnaceNum).SchedPtr) > 0.0 &&
        ((GetCurrentScheduleValue(state, state.dataFurnaces->Furnace(FurnaceNum).FanAvailSchedPtr) > 0.0 || state.dataHVACGlobal->TurnFansOn) &&
         !state.dataHVACGlobal->TurnFansOff)) {

    // IF the furnace is scheduled on or nightime cycle overrides fan schedule. Uses same logic as fan.
    if (GetCurrentScheduleValue(state, state.dataFurnaces->Furnace(FurnaceNum).SchedPtr) > 0.0 &&
        ((GetCurrentScheduleValue(state, state.dataFurnaces->Furnace(FurnaceNum).FanAvailSchedPtr) > 0.0 || state.dataHVACGlobal->TurnFansOn) &&
         !state.dataHVACGlobal->TurnFansOff)) {


    // set the mass flow rates from the input volume flow rates
    if (OAFrac > 0.0 || (ZoneCompTurnFansOn && !ZoneCompTurnFansOff)) { // fan is available
        OutAirUnit(OAUnitNum).OutAirMassFlow = RhoAir * OAFrac * OutAirUnit(OAUnitNum).OutAirVolFlow;
    } else {
        OutAirUnit(OAUnitNum).OutAirMassFlow = 0.0;
    }

    // if the terminal unit and fan are scheduled on then set flow rate
    if (GetCurrentScheduleValue(state, state.dataHVACVarRefFlow->VRFTU(VRFTUNum).SchedPtr) > 0.0 &&
        (GetCurrentScheduleValue(state, state.dataHVACVarRefFlow->VRFTU(VRFTUNum).FanAvailSchedPtr) > 0.0 ||
         state.dataHVACGlobal->ZoneCompTurnFansOn) &&
        !state.dataHVACGlobal->ZoneCompTurnFansOff) {

    if (ScheduleManager::GetCurrentScheduleValue(state, this->m_SysAvailSchedPtr) > 0.0 && ((FanOn || turnFansOn) && !turnFansOff)) { // UnitarySystem


// turn off logic

    if (GetCurrentScheduleValue(state, state.dataPowerInductionUnits->PIU(PIUNum).SchedPtr) <= 0.0) PriOn = false;
    if ((GetCurrentScheduleValue(state, state.dataPowerInductionUnits->PIU(PIUNum).FanAvailSchedPtr) <= 0.0 || PIUTurnFansOff) && !PIUTurnFansOn)
        UnitOn = false;

    if ((std::abs(state.dataUnitVentilators->QZnReq) < DataHVACGlobals::SmallLoad) ||
        (state.dataZoneEnergyDemand->CurDeadBandOrSetback(ZoneNum)) ||
        (ScheduleManager::GetCurrentScheduleValue(state, unitVent.SchedPtr) <= 0) ||
        ((ScheduleManager::GetCurrentScheduleValue(state, unitVent.FanAvailSchedPtr) <= 0 && !ZoneCompTurnFansOn) || ZoneCompTurnFansOff)) {
        // Unit is off or has no load upon it; set the flow rates to zero and then

Reply:
I just ran into this again. The night cycle avail manager object set up is that the coils need to be available when the fans turn on. The avail manager overrides the fan avail schedule. But the coils won't turn on if they are not available. Your right that TurnFansOn overrides the fan schedule (constant or cycling fan op mode).